specification/cognitiveservices/data-plane/Language/stable/2021-10-01/questionanswering-authoring.json

```diff
@@ -30,7 +30,7 @@
     "/query-knowledgebases/projects": {
       "get": {
         "summary": "Gets all projects for a user.",
-        "operationId": "QuestionAnsweringProjects_ListProjects",
+        "operationId": "QuestionAnsweringAuthoring_ListProjects",
```

Comment:
Changing the operation IDs will change the client and method names for the SDKs.  Given that this spec is marked as stable, we can't allow such breaking changes (especially if any SDKs have already been released).

Reply:
@jhendrixMSFT - These have been taken care of from the SDK side by @AhmedLeithy. The reason for making this change is that the REST API Docs are directly generated using the swaggers and there's no way to override the naming there.

@heaths - Could you take a look at this PR. Thanks!

Reply:
I have added transforms to accommodate these changes in a [PR here](https://github.com/Azure/azure-sdk-for-net/blob/9bd1bffaa936d5a09f5d947376204f785441c9d7/sdk/cognitivelanguage/Azure.AI.Language.QuestionAnswering/src/autorest.md).

Reply:
@AhmedLeithy it would be far more readable and maintainable if you use `rename-operation`: https://github.com/Azure/autorest/blob/6caa31c18ff03173a8f4a945bfe0306c94463aa8/docs/generate/built-in-directives.md#rename-operation. That said, it does create a dependency on merging this PR which I'll review regardless.

Reply:
Can you please clarify how this fixes the docs?  While it's great that you used a transform to avoid the breaking changes in the .NET SDK, you'd need to apply (and maintain) the same transforms to all languages that you support.  This just seems very fragile and error-prone.

CC @mikekistler

Reply:
@jhendrixMSFT , @mikekistler  - The [REST API docs](https://docs.microsoft.com/en-us/rest/api/cognitiveservices/questionanswering/question-answering) are auto-generated from the swaggers, and the API groupings/folder structure is picked up using the OperationIds.

Currently it looks something like this -

![image](https://user-images.githubusercontent.com/36928710/148272640-e65dd9d0-5f13-4877-87c1-e0a5bae7bf2e.png)

Customers have reported that they're confused by the current naming. As there's no way to override this at the docs level, we have to do it in the swagger itself. Hope this helps, please let me know if you have further questions.

Reply:
> As there's no way to override this at the docs level, we have to do it in the swagger itself.

Have you talked with the Docs team? After discussing it with Joel, I agree that transforming our swaggers that affect customers' production code for the sake of docs seems backward.

That said, I'm less concerned about non-autorest generated code. How we use `operationId` seems very much Microsoft-specific (rather, autorest-specific).

Reply:
@heaths  - I have discussed with the folks working on the documentation and they don't have any way to override the swagger data. I'm making this change only because the `operationId ` field is something which does not affect the API contract in anyway and is used internally. It's not a breaking change for the API contract.

@jhendrixMSFT , @mikekistler  -  Please let me know if that's okay.

Reply:
If we have not generated a client from the prior version of the API definition, then it should be safe to change the `operationId`, and I consider that far better than using transforms in the README.

Reply:
In this case it seems that there is an impasse. It is clearly unacceptable to publish an SDK with one set of names and use a different set of names in the docs.  For this change to be acceptable would require that the SDKs be regenerated to use the new names.  So if this is what is desired, the case needs to be made to the Arch Board and the Arch Board must agree to this.